ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -101,7 +102,13 @@ export function WorkflowLogsViewer({workflow, nodeId, initialPodName, container,
     useEffect(() => {
         setError(null);
         setLoaded(false);
-        const source = services.workflows.getContainerLogs(workflow, podName, nodeId, selectedContainer, grep, archived).pipe(
+
+        // if no node id is set (for example, when no node is selected), then use the node id of of the pod.
+        if (archived && !selectedNodeId && podNamesToNodeIDs.get(podName)) {
+            setNodeId(podNamesToNodeIDs.get(podName));
+        }
```

Comment:
I believe this case should no longer be possible now.

it's also recursive -- `selectedNodeId` is both an argument / `useEffect` dependency and set here. I think that technically works due to the conditional, but that's also an anti-pattern

Reply:
no longer needed nor possible after [the derivation](https://github.com/argoproj/argo-workflows/pull/12591#discussion_r1563827615)